test/test_libs.py

```diff
@@ -3930,6 +3939,127 @@ def test_chance_not_implemented(self):
             OpenSpielEnv("bridge")
 
 
+# NOTE: Each of the registered envs are around 180 MB, so only test a few.
+_mlagents_registered_envs = [
+    "3DBall",
+    "StrikersVsGoalie",
+]
+
+
+@pytest.mark.skipif(not _has_unity_mlagents, reason="mlagents_envs not found")
+class TestUnityMLAgents:
+    @mock.patch("mlagents_envs.env_utils.launch_executable")
+    @mock.patch("mlagents_envs.environment.UnityEnvironment._get_communicator")
+    def test_env(self, mock_communicator, mock_launcher):
+        from mlagents_envs.mock_communicator import MockCommunicator
+
+        mock_communicator.return_value = MockCommunicator(
+            discrete_action=False, visual_inputs=0
+        )
+        env = UnityMLAgentsEnv(" ")
+        try:
+            check_env_specs(env)
+        finally:
+            env.close()
+
+    @mock.patch("mlagents_envs.env_utils.launch_executable")
+    @mock.patch("mlagents_envs.environment.UnityEnvironment._get_communicator")
+    def test_wrapper(self, mock_communicator, mock_launcher):
+        from mlagents_envs.environment import UnityEnvironment
+        from mlagents_envs.mock_communicator import MockCommunicator
+
+        mock_communicator.return_value = MockCommunicator(
+            discrete_action=False, visual_inputs=0
+        )
+        env = UnityMLAgentsWrapper(UnityEnvironment(" "))
+        try:
+            check_env_specs(env)
+        finally:
+            env.close()
+
+    @mock.patch("mlagents_envs.env_utils.launch_executable")
+    @mock.patch("mlagents_envs.environment.UnityEnvironment._get_communicator")
+    def test_rollout(self, mock_communicator, mock_launcher):
+        from mlagents_envs.environment import UnityEnvironment
+        from mlagents_envs.mock_communicator import MockCommunicator
+
+        mock_communicator.return_value = MockCommunicator(
+            discrete_action=False, visual_inputs=0
+        )
+        env = UnityMLAgentsWrapper(UnityEnvironment(" "))
+        try:
+            env.rollout(
+                max_steps=500, break_when_any_done=False, break_when_all_done=False
+            )
+        finally:
+            env.close()
+
+    @pytest.mark.unity_editor
+    def test_with_editor(self):
+        print("Please press play in the Unity editor")  # noqa: T201
```

Comment:
what's this?

Reply:
When `UnityMLAgentsEnv` is not given a path to a file or the name of a registered env, it waits for the user to press the play button in the unity editor or run a mlagents-compatible binary. Since we can't easily do that in CI, I figured we can just skip the test by default. To run the test, the cmd line arg `--unity_editor` has to be given. We could remove this test, but I found it useful while working on this PR to have a test that connects to whichever environment I have open in the unity editor

Reply:
gotcha!